File: Hanoi.py
def Hanoi (discos,origen=1,auxiliar=2,destino=3):
    
    if discos==1:
        print (origen,"a",destino)
        
    else:
        Hanoi(discos-1,origen,destino,auxiliar)
        print(origen,"a",destino)
        Hanoi (discos-1,auxiliar,origen,destino)
    return

Hanoi(3)

File: Hanoi.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TorreDeHanoi {
    class Hanoi {
        static void Main(string[] args) {

            Hanoi h = new Hanoi();

            char inicial='1';
            char auxiliar='2';
            char final='3';

            int n;

            String a;

            Console.WriteLine("Número de discos: ");
            a = Console.ReadLine();

            n = int.Parse(a);

            Console.WriteLine("Los movimientos a realizar son: ");
            h.hanoi(n,inicial,auxiliar,final);
            
        }

        void hanoi(int n, char c, char a, char f) {

            if(n==1){

            Console.WriteLine(c + " -> " + f);

            }else{

            hanoi(n-1,c,f,a);

            Console.WriteLine(c + " -> " + f);

            hanoi(n-1,a,c,f);
            
            }
            
        }

    }

}